### DISCUSSION TITLE
ERROR : can't read "tablename": no such variable , when running command dbset db pg (Postgres)

### DISCUSSION THREAD

**Discussion by noopur100** [TPC-Council/HammerDB#417] [Q&A] [answered]
I am new to Hammerdb and trying to run command "dbset db pg " but it gives me below error, any idea what can be done to resolve it ?

hammerdb>dbset db pg
Database set to PostgreSQL
can't read "tablename": no such variable
    while executing
"puts "Error creating $tablename table in $sqlitedb : $message""
    invoked from within
"if [catch {hdb eval $sqlcmd} message ] {
                puts "Error creating $tablename table in $sqlitedb : $message"
                return
       ..."
    invoked from within
"if [catch {sqlite3 hdb $sqlitedb} message ] {
        puts "Error initializing SQLite database : $message"
        return
    } else {
        catch {..."
    (procedure "SQLiteUpdateKeyValue" line 4)
    invoked from within
"SQLiteUpdateKeyValue "generic" "benchmark" "rdbms" $rdbms"
    (procedure "dbset" line 30)
    invoked from within
"dbset db pg"

**Comment by sm-shaw** [comment, marked as answer]
From v4.5 the parameters are stored in a series of SQLite files - it looks like you are having issues with the permissions to create these files.
By default, they will be created in a temp directory. You don't mention what OS you are using, so using Linux as an example after starting HammerDB you should see a series of files in /tmp as follows:

```
-rw-r--r-- 1 postgres postgres 57344 Aug 18 09:39 generic.db
-rw-r--r-- 1 postgres postgres 28672 Aug 18 09:39 database.db
-rw-r--r-- 1 postgres postgres 16384 Aug 18 09:39 oracle.db
-rw-r--r-- 1 postgres postgres 16384 Aug 18 09:39 mssqlserver.db
-rw-r--r-- 1 postgres postgres 16384 Aug 18 09:39 db2.db
-rw-r--r-- 1 postgres postgres 16384 Aug 18 09:39 mysql.db
-rw-r--r-- 1 postgres postgres 16384 Aug 18 09:39 postgresql.db
-rw-r--r-- 1 postgres postgres 16384 Aug 18 09:39 mariadb.db
```

the location is set in the config/generic.xml file as follows:

```
  <sqlitedb>
        <sqlitedb_dir>TMP</sqlitedb_dir>
   </sqlitedb>
```

If you have security settings that is restricting the use of SQLite or opening database files this would result in the sort of problem you are seeing.

What you should do is either set sqlitedb_dir to a directory you have read and write permissions or alternatively set the TMP environment variable to the name of the directory you want to use without changing generic.xml. It will have the same effect.

eg creating /opt/sqlitefiles with the correct permissions and doing export TMP=/opt/sqlitefiles before starting will create the .db files in /opt/sqlitefiles.

**Reply by noopur100** [reply]
Thank you so much, it worked !!!

**Reply by sm-shaw** [reply]
Great, Also to note that if you want to reset the data at any time you can delete these .db files and HammerDB will recreate them on startup. 
Their main function is to retain the settings you have made across restarts.
